docs/apis/plugintypes/format/index.mdx

```diff
@@ -0,0 +1,613 @@
+
+If your plugin needs a big UI change, altering a considerable number of the course elements at different levels (course, section and/or activity), the previous approach is not recommended as it may require more code refactoring in future versions.
+
+To keep your plugin more stable through time the best approach is to override the mustache blocks at the inner parts of the mustache files structure instead of at the main elements. It will require more code but the final result will be more stable.
+
+<details>
+  <summary>View example</summary>
+  <div>
+
+Let's say your format requires overriding the activity visibility badges as in the previous scenario example. Apart from the three main templates, the plugin must create several new template overrides until it reacher the activity badges one:
+
+CM item main file: format_pluginname/local/content/section/cmitem template:
+
+```handlebars
+{{! include the original course format template block }}
+{{< core_courseformat/local/content/section/cmitem }}
+    {{$ core_courseformat/local/content/cm }}
+        {{> format_pluginname/local/content/cm }}
+    {{/ core_courseformat/local/content/cm }}
+{{/ core_courseformat/local/content/section/cmitem }}
+```
+
+The content/cm template:
+
+```handlebars
+{{< core_courseformat/local/content/cm/activity }}
+    {{$ core_courseformat/local/content/cm/badges }}
+        {{> format_pluginname/local/content/cm/badges }}
+    {{/ core_courseformat/local/content/cm/badges }}
+{{/ core_courseformat/local/content/cm/activity }}
+```
+
+The content/cm/badges will contain only the overridden HTML.
+
+Apart from the templates files, the plugin could also provide overridden output classes to ensure that future versions will remain compatible if new ajax partials are required:
+
+In the example the extra output classes can look like:
+
+```php tile="format_pluginname\output\local\content\cm class"
+<?php
+namespace format_pluginname\output\courseformat\content;
+
+class cm extends \core_courseformat\output\local\content\cm {
+    public function get_template_name(\renderer_base $renderer): string {
+        return 'format_pluginname/local/content/cm;
+    }
+}
+```
+
+```php title="format_pluginname\output\local\content\cm\activity class"
+<?php
+namespace format_pluginname\output\courseformat\content\cm;
+
+class activity extends \core_courseformat\output\local\content\cm\activity {
+    public function get_template_name(\renderer_base $renderer): string {
+        return 'format_pluginname/local/content/cm/activity;
+    }
+}
+```
+
+```php title="format_pluginname\output\local\content\cm\badge class"
+<?php
+namespace format_pluginname\output\courseformat\content\cm;
+
+class badge extends \core_courseformat\output\local\content\cm\badge {
+    public function get_template_name(\renderer_base $renderer): string {
+        return 'format_pluginname/local/content/cm/badge;
+    }
+}
+```
+
+  </div>
+</details>
+
+Pros of that approach:
```

Comment:
Suggest "Benefits of this approach" and "Negatives to this approach" - Pros/Cons are quite colloquial.

Reply:
Well spotted. I fix it.